src/Microsoft.Health.Fhir.CosmosDb/Features/Search/Queries/QueryBuilder.cs

```diff
@@ -47,14 +48,18 @@ public QueryBuilderHelper()
                 _queryHelper = new QueryHelper(_queryBuilder, _queryParameterManager, SearchValueConstants.RootAliasName);
             }
 
-            public QueryDefinition BuildSqlQuerySpec(SearchOptions searchOptions, IReadOnlyList<IncludeExpression> includes)
+            public QueryDefinition BuildSqlQuerySpec(SearchOptions searchOptions, IReadOnlyList<IncludeExpression> includes, bool idOnly)
```

Comment:
>idOnly [](start = 130, length = 6)

I would push that into SearchOptions class as property.
I think it looks better and you don't have to stretch function parameters.

Reply:
I had this though also, but `SearchOptions` is in Core and applicable to all DBs, this seemed specific to this implementation. I'll think about it some more

Reply:
Added `QueryBuilderOptions` to encapsulate these